src/main/java/net/dv8tion/jda/api/interactions/commands/build/CommandData.java

```diff
@@ -142,7 +142,8 @@ public CommandData addOptions(@Nonnull OptionData... options)
         Checks.noneNull(options, "Option");
         Checks.check(options.length + this.options.length() <= 25, "Cannot have more than 25 options for a command!");
         Checks.check(allowOption, "You cannot mix options with subcommands/groups.");
-        allowSubcommands = allowGroups = false;
+        if (options.length > 0)
+            allowSubcommands = allowGroups = false;
```

Comment:
You should just return if length = 0, also this check should be before the mixing of options check is done.